App/VSTS/FollowButton.tsx

```diff
@@ -0,0 +1,66 @@
+/// <reference path="../../office.d.ts" />
+import * as React from 'react';
+import { Provider } from 'react-redux';
+import { changeFollow } from '../actionsEZ';
+import { connect } from 'react-redux';
+import { FollowStateTypes } from '../statesEZ';
+
+let buttonText: string = '';
+let isDisabled: boolean = false;
+let iconType: string = 'ms-Icon ms-Icon--eye';
+
+interface IFollowButtonProps {
+  dispatch?: any;
+  followState?: FollowStateTypes;
+}
+
+function mapStateToProps(state: any): IFollowButtonProps {
+  return {
+    followState: state.quickActionState.followState,
+  };
+}
+
+@connect(mapStateToProps)
+export class FollowButton extends React.Component<IFollowButtonProps, {}> {
+  public render(): React.ReactElement<Provider> {
+    console.log('got to followbutton, followstate: ' + this.props.followState);
+    switch (this.props.followState) {
+      case FollowStateTypes.Followed:
+        buttonText = 'Following';
+        break;
+      case FollowStateTypes.Unfollowed:
+        buttonText = 'Follow';
+        break;
+      default:
+        break;
+    }
+
+    return(
+      <div>
+        <button className='ms-Button' id='follow' disabled={isDisabled} onClick = {this.handleClick}>
+        <span className={iconType} ></span>
+        {'   ' + buttonText}
+        </button>
+        <h1></h1>
+      </div>
+    );
+  }
+
+  private handleClick: any = (event) => {
+    switch (this.props.followState) {
+      case FollowStateTypes.Followed:
+        this.props.dispatch(changeFollow(FollowStateTypes.Unfollowed));
+        buttonText = 'Following';
+        isDisabled = false;
```

Comment:
isDisabled looks to be representing an internal state here and we shouldn't be doing this. isDisabled is only true when the state is set to Request. You can just check for it inside of render itself.